src/core/pratt.ml

```diff
@@ -0,0 +1,113 @@
+(** This module defines functions needed by the Pratt parser of the Pratter
+    library. *)
+
+open Syntax
+
+module Pratt : sig
+  val parse : Sig_state.t -> Env.t -> p_term -> p_term
+end = struct
+
+  open Lplib
+  open Extra
+  open Pos
+
+  (** [find_qid ss env qid] fetches the qualified identifier associated to
+      [qid] if [qid] is the identifier of an infix operator defined in
+      signature state [ss]. If [qid] is defined in environment [env], then it
+      designates a variable, and is thus returned as-is. *)
+  (* REVIEW: this function relies on a lot of invariants and performs many
+     [Map.find_opt]s. Things may be optimised by modifying the data structure
+     of the signature state, notably making the qualified identfier to which
+     operators are linked available in the [unops] field of the signature
+     state. *)
+  let find_qid : Sig_state.t -> Env.t -> qident -> qident =
+    fun ss env ({elt=(mp, s); _} as qid) ->
+    try
+      if mp <> [] then raise Not_found;
+      ignore (Env.find s env);
+      qid
```

Comment:
Why not having find_qid return a `qident option` and here, in the case of a variable, replace `ignore (Env.find s env); qid` by `Some (Env.find s env)`?

Reply:
Well `Env.find s env` returns a `tvar`, not a `qident`, so your `Some (Env.find s env)` is ill-typed, isn't it?

Reply:
Ah, right. It should be `ignore (Env.find s env); None`.